File: stack/evaluate-reverse-polish-notation/evaluate_reverse_polish_notation.py
class Solution:
    def evalRPN(self, tokens) -> int:
        stack = []
        for token in tokens:
            if token == '+':
                n1 = stack.pop()
                n2 = stack.pop()
                stack.append(n2 + n1)
            elif token == '-':
                n1 = stack.pop()
                n2 = stack.pop()
                stack.append(n2 - n1)
            elif token == '*':
                n1 = stack.pop()
                n2 = stack.pop()
                stack.append(n2 * n1)
            elif token == '/':
                n1 = stack.pop()
                n2 = stack.pop()
                if n2 // n1 < 0:
                    stack.append(-(abs(n2) // abs(n1)))
                else:
                    stack.append(n2 // n1)
            else:
                stack.append(int(token))
            
        return stack[0]


s = Solution()
print(s.evalRPN(["10","6","9","3","+","-11","*","/","*","17","+","5","+"]))

File: stack/evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.ts
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function (tokens: string[]) {
  let stack: number[] = [];
  for (let token of tokens) {
    let t = Number(token);
    if (isNaN(t)) {
      let n1: number = stack.pop() || 0;
      let n2: number = stack.pop() || 0;
      switch (token) {
        case '+':
          stack.push(n2 + n1);
          break;
        case '-':
          stack.push(n2 - n1);
          break;
        case '*':
          stack.push(n2 * n1);
          break;
        case '/':
          let d = n2 / n1;
          if (d < 0) {
            d = -Math.floor(-d);
          } else {
            d = Math.floor(d);
          }
          stack.push(d);
          break;
      }
    } else {
      stack.push(t);
    }
  }
  return stack[0];
};
